### PR TITLE
fix: osmosis staking provider name

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- Makes sure we always use `Cosmos/Osmosis` as the provider name depending on the current opportunity asset
- Coincidentally decouples `DefiProvider` from the view-layer with `makeDefiProviderDisplayName` e.g the display name and the enum may not be different 
- Fixes a bunch of types in the process

This also paves the way for provider metadata. As discussed with big @reallybeard yesterday, we will need to have provider metadata e.g name, icon src, and possibly more properties. Still not sure what's the best place for it yet, but by leveraging grep fu, we will be able to access the provider metadata by `DefiProvider` and do `providerMetadata.name` instead of using this method in-place.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

`<StakingTable />` now has a dependency on `assets`, watch out for re-render perf

## Testing

- Cosmos staking still shows up as `Cosmos`
- Osmosis staking shows up as `Osmosis`
- Other provider names show no regressions

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

Screenshot 2023-02-17 at 10 56 53
Screenshot 2023-02-17 at 10 56 42
Screenshot 2023-02-17 at 10 56 22

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3859** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3859" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3859?utm_source=stack-comment).

File: src/components/StakingVaults/StakingTable.tsx [hunk 1]
```diff
@@ -20,6 +24,7 @@ type StakingTableProps = {
 type RowProps = Row<EarnOpportunityType>
 
 export const StakingTable = ({ data, onClick, showTeaser }: StakingTableProps) => {
+  const assets = useAppSelector(selectAssets)
```
Comment (0xdef1cafe) [inline src/components/StakingVaults/StakingTable.tsx:27]: we can avoid this, and passed in `row.original.assetId`, and have `makeDefiProviderDisplayName` do the suggested thing below

File: src/state/slices/opportunitiesSlice/utils.ts [hunk 2]
```diff
@@ -114,3 +115,22 @@ export const makeOpportunityIcons = ({
   opportunity.icon
     ? [opportunity.icon]
     : opportunity.underlyingAssetIds.map(assetId => assets[assetId]?.icon).map(icon => icon ?? '')
+
+export const makeDefiProviderDisplayName = ({
+  provider,
+  asset,
+}: {
+  provider: DefiProvider
+  asset: Asset | undefined
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/utils.ts:124]: - pass in `AssetId` then `const asset = store.getState.assets.byId[assetId]`

**Comment by 0xdef1cafe** [comment]
@gomesalexandre if you agree with [ff6035c](https://github.com/shapeshift/web/pull/3859/commits/ff6035ce2a99c8435113a7cf39feb24398221d9b) feel free to merge, i'll approve yours if you approve mine

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by gomesalexandre** [comment]
> @gomesalexandre if you agree with [ff6035c](https://github.com/shapeshift/web/pull/3859/commits/ff6035ce2a99c8435113a7cf39feb24398221d9b) feel free to merge, i'll approve yours if you approve mine

LGTM! Not being reactive on assets is actually a lot better since `data` e.g earn opportunities data, will by itself be reactive on assets population

**Pull request merged**